### ISSUE TITLE
Address various a11y issues in Button docs

### ISSUE THREAD

**Issue by rolfsmeds** [vaadin/docs#2606]
- [ ] Sample with images: Add aria-labels
- [ ] Prevent multiple clicks: Add note that focus should be moved to an appropriate element when the button becomes disabled. Also add an aria-live element that announces that progress is ongoing (e.g. the progress-bar itself).
- [ ] Autofocus: Add note that moving focus programmatically can be confusing, so should be used with care.

---

If you think this issue is important, add a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to help the community and maintainers prioritize this issue.
